WordPress/Sniffs/Theme/DeprecatedWPFunctionsSniff.php

```diff
@@ -0,0 +1,341 @@
+			array( 'get_broken_themes' => 'wp_get_themes( array( \'errors\' => true )', '3.4' ),
+			array( 'current_theme_info' => 'wp_get_theme()', '3.4' ),
+
+			array( '_insert_into_post_button' => '', '3.5' ),
+			array( '_media_button' => '', '3.5' ),
+			array( 'get_post_to_edit' => 'get_post()', '3.5' ),
+			array( 'get_default_page_to_edit' => 'get_default_post_to_edit()', '3.5' ),
+			array( 'wp_create_thumbnail' => 'image_resize()', '3.5' ),
+
+			array( 'wp_nav_menu_locations_meta_box' => '', '3.6' ),
+
+			array( 'the_attachment_links' => '', '3.7' ),
+			array( 'wp_update_core' => 'new Core_Upgrader()', '3.7' ),
+			array( 'wp_update_plugin' => 'new Plugin_Upgrader()', '3.7' ),
+			array( 'wp_update_theme' => 'new Theme_Upgrader()', '3.7' ),
+			array( '_search_terms_tidy' => '', '3.7' ),
+			array( 'get_blogaddress_by_domain' => '', '3.7' ),
+
+			array( 'get_screen_icon' => '', '3.8' ),
+			array( 'screen_icon' => '', '3.8' ),
+			array( 'wp_dashboard_incoming_links' => '', '3.8' ),
+			array( 'wp_dashboard_incoming_links_control' => '', '3.8' ),
+			array( 'wp_dashboard_incoming_links_output' => '', '3.8' ),
+			array( 'wp_dashboard_plugins' => '', '3.8' ),
+			array( 'wp_dashboard_primary_control' => '', '3.8' ),
+			array( 'wp_dashboard_recent_comments_control' => '', '3.8' ),
+			array( 'wp_dashboard_secondary' => '', '3.8' ),
+			array( 'wp_dashboard_secondary_control' => '', '3.8' ),
+			array( 'wp_dashboard_secondary_output' => '', '3.8' ),
+
+			array( 'rich_edit_exists' => '', '3.9' ),
+			array( 'default_topic_count_text' => '', '3.9' ),
+			array( 'format_to_post' => '', '3.9' ),
+			array( 'get_current_site_name' => 'get_current_site()', '3.9' ),
+			array( 'wpmu_current_site' => '', '3.9' ),
+			array( '_relocate_children' => '', '3.9' ),
+
+			array( 'get_all_category_ids' => 'get_terms()', '4.0' ),
+			array( 'like_escape' => 'wpdb::esc_like()', '4.0' ),
+			array( 'url_is_accessable_via_ssl' => '', '4.0' ),
+
+			array( 'prepare_control' => '', '4.1' ),
+			array( 'add_tab' => '', '4.1' ),
+			array( 'remove_tab' => '', '4.1' ),
+			array( 'print_tab_image' => '', '4.1' ),
+
+			array( 'setup_widget_addition_previews' => 'customize_dynamic_setting_args', '4.2' ),
+			array( 'prepreview_added_sidebars_widgets' => 'customize_dynamic_setting_args', '4.2' ),
+			array( 'prepreview_added_widget_instance' => 'customize_dynamic_setting_args', '4.2' ),
+			array( 'remove_prepreview_filters' => 'customize_dynamic_setting_args', '4.2' ),
+
+			array( 'wp_get_http' => 'WP_Http', '4.4' ),
+
+			array( 'is_comments_popup' => '', '4.5' ),
+			array( 'add_object_page' => 'add_menu_page', '4.5' ),
+			array( 'add_utility_page' => 'add_menu_page', '4.5' ),
+			array( 'get_comments_popup_template' => '', '4.5' ),
+			array( 'comments_popup_script' => '', '4.5' ),
+			array( 'popuplinks' => '', '4.5' ),
+			array( 'get_currentuserinfo' => 'wp_get_current_user', '4.5' ),
+		);
+		foreach ( $checks as $alt => $check ) {
```

Comment:
By changing the format of the array, here you wouldn't need to loop through all of the functions (which will become terrible for performance). Instead, you can just do something like:

```php
if ( isset( $checks[ trim( $token['content'] ) ] ) ) {
     //...
}
```

No need for a loop! 😄